### PR TITLE
Pause function

### PR DESCRIPTION
**Pull request by pmw9027**
insert pause toolbar item

### CODE REVIEW

**Comment by youknowone** [comment]
Because loading text is merged in other PR, rebase will be appricated.

**Comment by youknowone** [comment]
I found it - you should disable `autovalidate` property of the ToolbarItem in the storyboard.
Then it will allow you to change isEnabled manually.

**Comment by youknowone** [comment]
Will you rebase the this PR?

File: Say/Say.swift [hunk 1]
```diff
@@ -17,12 +17,14 @@ class SayAPI: NSObject {
     var voice: VoiceAPI?
     
     var outputFile: String? = nil
-    
+    var speechToolBar:NSToolbarItem? = nil
```
Comment (youknowone) [inline Say/Say.swift]: Is this used anywhere?

File: Say/Say.swift [hunk 2]
```diff
@@ -17,12 +17,14 @@ class SayAPI: NSObject {
...
     override var description: String {
         get {
             return "<Say: '\(self.text)'>"
         }
     }
+
```
Comment (youknowone) [inline Say/Say.swift]: redundant whitespace

File: Say/Say.swift [hunk 3]
```diff
@@ -37,7 +39,14 @@ class SayAPI: NSObject {
         self.voice = voice
         super.init()
     }
+    /*
```
Comment (youknowone) [inline Say/Say.swift]: meaningless comments

File: Say/ViewController.swift [hunk 4]
```diff
@@ -36,21 +40,28 @@ class MainWindow: NSWindow {
             }
             return iconData
         }
-        
+        speechToolbarItem.autovalidates = false
```
Comment (youknowone) [inline Say/ViewController.swift]: You can change this value in Interface builder.

File: Say/ViewController.swift [hunk 5]
```diff
@@ -9,12 +9,16 @@
...
-
 /// Main window of the application
 class MainWindow: NSWindow {
     @IBOutlet var speechToolbarItem: NSToolbarItem! = nil
+    @IBOutlet var pauseToolbarItem: NSToolbarItem! = nil
     @IBOutlet var exportToolbarItem: NSToolbarItem! = nil
     @IBOutlet var openToolbarItem: NSToolbarItem! = nil
+    @IBOutlet var stopToolbarItem: NSToolbarItem! = nil
+    //@IBOutlet var statusItem: NSStatusItem! = nil
+    //@IBOutlet var statusMenu: NSMenu! = nil
```
Comment (aweekj) [inline Say/ViewController.swift]: Do we need to remain these comments? It looks unnecessary to remain these two lines ;(

File: SayKit [hunk 6]
```diff
@@ -0,0 +1 @@
+Subproject commit 2054a7e6c36e8d00bc99ff6ce06c1fe833e098f3
```
Comment (youknowone) [inline SayKit]: SayKit shouln't revive.

File: Say/ViewController.swift [hunk 7]
```diff
@@ -148,19 +162,48 @@ class ViewController: NSViewController {
...
+    @IBAction func say(_ sender: NSToolbarItem) {
+        
+        if !say.isplaying(){
+            
+            if self.pause{
+            
+                say.continueSpeeking()
+            }
+            else{
+                //sender.isEnabled = false
```
Comment (youknowone) [inline Say/ViewController.swift]: let's sweep out non-using codes

File: Say/ViewController.swift [hunk 8]
```diff
@@ -148,19 +162,48 @@ class ViewController: NSViewController {
...
+                say = SayAPI(text: self.textForSpeech, voice: self.selectedVoice)
+                say.speechToolBar = sender
+                say.play(false)
+                
+            }
+            //sender.isEnabled = true
+        }
+        
+    }
```
Comment (aweekj) [inline Say/ViewController.swift:177]: I think there is too many empty lines here. How about to clean up these wasted spaces?

**Review by youknowone — CHANGES_REQUESTED** [review]

File: Say/Say.swift [hunk 9]
```diff
@@ -64,7 +73,24 @@ class SayAPI: NSObject {
      */
     func play(_ waitUntilDone: Bool) {
         self.speechSynthesizer.setVoice(self.voice?.identifier)
-        self.speechSynthesizer.startSpeaking(text);
+        self.speechSynthesizer.startSpeaking(text)
+    }
+    
+    func continueSpeeking(){
```
Comment (youknowone) [inline Say/Say.swift]: code style `() {`

**Review by youknowone — CHANGES_REQUESTED** [review]

File: Say/ViewController.swift [hunk 10]
```diff
@@ -148,10 +159,35 @@ class ViewController: NSViewController {
...
     }
     
-    @IBAction func say(_ sender: NSControl) {
-        sender.isEnabled = false
-        SayAPI(text: self.textForSpeech, voice: self.selectedVoice).play(false)
-        sender.isEnabled = true
+    
+    @IBAction func say(_ sender: NSToolbarItem) {
+        
+        if !say.isplaying(){
```
Comment (youknowone) [inline Say/ViewController.swift]: `() {`

**Review by youknowone — CHANGES_REQUESTED** [review]

File: Say/ViewController.swift [hunk 11]
```diff
@@ -148,10 +159,35 @@ class ViewController: NSViewController {
...
+            
+            if self.pause{
+            
+                say.continueSpeeking()
+            }
```
Comment (youknowone) [inline Say/ViewController.swift]: we use `} else {` widely in this project

File: SayKit (hunk 6 above)
Comment (youknowone) [reply SayKit]: You didn't remove SayKit yet.

**Review by youknowone — CHANGES_REQUESTED** [review]

File: Say/ViewController.swift (hunk 11 above)
Comment (Amanida) [reply Say/ViewController.swift]: say.continueSpeeking() -> say.continue**Speaking()** 😄

Comment (youknowone) [reply Say/ViewController.swift]: Nice catch. It looks a glitch of SayAPI

Comment (pmw9027) [reply Say/ViewController.swift]: oops..I modified

**Pull request merged**